evap/evaluation/models.py

```diff
@@ -284,7 +287,11 @@ def can_user_see_results(self, user):
         if user.is_staff:
             return True
         if self.state == 'published':
-            return self.can_publish_grades or self.is_user_contributor_or_delegate(user)
+            if self.is_user_contributor_or_delegate(user):
+                return True
+            if not self.can_publish_grades:
+                return False
+            return not self.is_private or user in self.participants.all()
```

Comment:
i thought, that is_private variable is supposed to restrict access if a condition is not met, and basically thats in the code
```
if private and not condition:
    restrict access
otherwise not
```
which i think is more straightforward than the current `give access exactly if not private and condition`

besides, i like the structure of handling three special cases, and then returning the default rather than handling two special cases and then returning the result of a third special case which has the default hidden in it. (only talking about the contents of the `if published` branch)